app/domain/authentication/common/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: ExtractResourceRestrictions.new,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id host_name account constraints authentication_request)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictions.new(@host_name))
+
+        resource_restrictions = extract_resource_restrictions
+        validate_resource_restrictions_configuration(resource_restrictions)
+        validate_request_matches_resource_restrictions(resource_restrictions)
+
+        @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictions.new)
+      end
+
+      private
+
+      def extract_resource_restrictions
+        @extract_resource_restrictions.call(
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id,
+          host_name:          @host_name,
+          account:            @account
+        )
+      end
+
+      def validate_resource_restrictions_configuration(resource_restrictions)
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictionsConfiguration.new)
+
+        @constraints.validate(
+          resource_restrictions: resource_restrictions.names
```

Comment:
maybe call this params `resource_restrictions_names`?

Reply:
Isn't it too long?
This string is all over the place... 😕

Reply:
i'm ok with the current name, just a suggestion. whichever you prefer

Reply:
I say, keep it shorter. thanks for the comment!